src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const statesArray = [];
+
+  let currentState = { ...state };
+
+  actions.forEach((action) => {
+    switch (action.type) {
+      case 'clear':
+        currentState = {};
+        break;
+
+      case 'addProperties':
+        currentState = { ...currentState, ...action.extraData };
+        break;
+
+      case 'removeProperties':
+        const copy = { ...currentState };
+        action.keysToRemove.forEach((key) => {
+          delete copy[key];
+        });
+
+        currentState = { ...copy };
```

Comment:
The reassignment of `currentState` to a new object created from `copy` is redundant. Since `copy` is already a new object, you can simply set `currentState` to `copy` without spreading it into another new object.